Formula/notmuch.rb

```diff
@@ -58,9 +67,15 @@ def install
     (prefix/"vim").install "vim/syntax"
 
     cd "bindings/python" do
-      system Formula["python@3.9"].opt_bin/"python3", *Language::Python.setup_install_args(prefix)
+      system "python3", *Language::Python.setup_install_args(prefix)
     end
 
+    venv = virtualenv_create(libexec, "python3")
+    venv.pip_install resources
+    venv.pip_install buildpath/"bindings/python-cffi"
+    pth_contents = "import site; site.addsitedir('#{libexec/site_packages}')\n"
+    (prefix/site_packages/"homebrew-notmuch2.pth").write pth_contents
```

Comment:
Not sure about this as it will also expose `cffi`, which could lead to unwanted behavior, though impact may be minimal.
```
>>> import notmuch2
>>> notmuch2.__file__
'/usr/local/Cellar/notmuch/0.35_1/libexec/lib/python3.9/site-packages/notmuch2/__init__.py'
>>> import cffi
>>> cffi.__file__
'/usr/local/Cellar/notmuch/0.35_1/libexec/lib/python3.9/site-packages/cffi/__init__.py'
```
Also not sure if we want to just install both bindings into venv.

Reply:
Not a fan of exposing `cffi` like this. If we have to do that then we may as well have a separate `cffi` formula, I think. As long as its users aren't too picky about the version, anyway.

Reply:
We have exposed some extra venv packages in other formulae that use pth files. One example is `woob` which has 14 other PyPI packages in venv. `ciphey` is worse with 27. 

Problem is that there doesn't seem to be an ideal way of providing bindings without exposing the other Python packages and not having to go Linux route of having formulae for every Python package.

Reply:
For now, just installed into `libexec` and added caveat on how users can make use of unlinked bindings rather than keep delaying PR.

Unless we decide to go the Linux distro route of packaging all Python packages as formulae, this isn't really going to get much better than either exposing random packagings or caveating formulae.